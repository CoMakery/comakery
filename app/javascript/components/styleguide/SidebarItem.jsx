import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import Icon from './Icon'

class SidebarItem extends React.Component {
  render() {
    const {
      className,
      iconLeftUrl,
      iconLeftName,
      iconRightName,
      text,
      selected,
      ...other
    } = this.props

    const classnames = classNames(
      'sidebar-item',
      (selected ? 'sidebar-item__selected' : ''),
      className
    )

    return (
      <React.Fragment>
        <div className="sidebar-item--wrapper" {...other}>
          <div className={classnames}>
            { (iconLeftUrl || iconLeftName) &&
              <span className="sidebar-item--icon-left">
                { iconLeftUrl !== '' &&
                  <img src={iconLeftUrl} />
                }
                { iconLeftName !== '' &&
                  <Icon name={iconLeftName} />
                }
              </span>
            }

            <span className="sidebar-item--text">
              {text}
            </span>

            { iconRightName !== '' &&
              <span className="sidebar-item--icon-right">
                <Icon name={iconRightName} />
              </span>
            }
          </div>
        </div>
      </React.Fragment>
    )
  }
}

SidebarItem.propTypes = {
  className    : PropTypes.string,
  iconLeftUrl  : PropTypes.string,
  iconLeftName : PropTypes.string,
  iconRightName: PropTypes.string,
  text         : PropTypes.string,
  selected     : PropTypes.bool
}
SidebarItem.defaultProps = {
  className    : '',
  iconLeftUrl  : '',
  iconLeftName : '',
  iconRightName: '',
  text         : 'sidebar item',
  selected     : false
}
export default SidebarItem
