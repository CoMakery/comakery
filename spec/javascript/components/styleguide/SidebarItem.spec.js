import React from 'react'
import { shallow, mount, render } from 'enzyme'
import SidebarItem from 'components/styleguide/SidebarItem'

describe('SidebarItem', () => {
    it('renders correctly without props', () => {
      const wrapper = shallow(<SidebarItem/>)

      expect(wrapper).toMatchSnapshot()
      expect(wrapper.exists('.sidebar-item')).toBe(true)
      expect(wrapper.exists('.sidebar-item__selected')).toBe(false)
      expect(wrapper.exists('.sidebar-item--wrapper')).toBe(true)
      expect(wrapper.exists('.sidebar-item--icon-left')).toBe(false)
      expect(wrapper.exists('.sidebar-item--text')).toBe(true)
      expect(wrapper.exists('.sidebar-item--icon-right')).toBe(false)
      expect(wrapper.find('.sidebar-item--text').text()).toBe('sidebar item')
    })

    it('renders correctly with custom className', () => {
      const wrapper = shallow(<SidebarItem className='__test' />)

      expect(wrapper.exists('.sidebar-item.__test')).toBe(true)
    })

    it('renders correctly with custom text', () => {
      const wrapper = shallow(<SidebarItem text='test' />)

      expect(wrapper.find('.sidebar-item--text').text()).toBe('test')
    })

    it('renders correctly with custom iconLeftUrl', () => {
      const wrapper = shallow(<SidebarItem iconLeftUrl='/ALERT.svg' />)

      expect(wrapper.exists('.sidebar-item--icon-left img')).toBe(true)
    })

    it('renders correctly with custom iconLeftName', () => {
      const wrapper = shallow(<SidebarItem iconLeftName='ALERT.svg' />)

      expect(wrapper.exists('.sidebar-item--icon-left Icon')).toBe(true)
    })

    it('renders correctly with custom iconRightName', () => {
      const wrapper = shallow(<SidebarItem iconRightName='ALERT.svg' />)

      expect(wrapper.exists('.sidebar-item--icon-right Icon')).toBe(true)
    })

    it('renders correctly when selected', () => {
      const wrapper = shallow(<SidebarItem selected={true} />)

      expect(wrapper.exists('.sidebar-item__selected')).toBe(true)
    })
})
