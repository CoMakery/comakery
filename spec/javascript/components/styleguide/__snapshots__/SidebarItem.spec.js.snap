// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarItem renders correctly without props 1`] = `
<Fragment>
  <div
    className="sidebar-item--wrapper"
  >
    <div
      className="sidebar-item"
    >
      <span
        className="sidebar-item--text"
      >
        sidebar item
      </span>
    </div>
  </div>
</Fragment>
`;
